burnman/mineral.py

```diff
@@ -64,6 +65,15 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
+            # Reset everything but params
+            for key, elem in self.__dict__.items():
+                if key != 'params':
+                    elem = float('nan')
+
```

Comment:
Well, the problem is that if we declare a new set_method, all of the old values of density, Vp, Vs are maintained. Some people might assume that properties are automatically recomputed. I'd rather not give anyone the change to misunderstand.

Even worse, if we try to set_state after set_method with the old P and T, your very useful catch to not recompute values if P and T is the same will end up confusing/misleading a lot of people.

If you don't like this method, I can suggest two alternatives:
1) Remove the P and T params only
2) Remove nothing, but output an additional warning, and store the old method as a temporary attribute that is deleted on the first new call of set_state.